zerver/lib/emoji.py

```diff
@@ -30,6 +35,46 @@ def check_valid_emoji(realm, emoji_name):
     # type: (Realm, Text) -> None
     emoji_name_to_emoji_code(realm, emoji_name)
 
+def emoji_code_is_valid(realm, emoji_code, emoji_type):
+    # type: (Realm, Text, Text) -> bool
+    # For a given realm and emoji type, this function determines whether the
+    # current emoji code is valid or not. In case of a deactivated realm emoji,
+    # its emoji code doesn't count towards a valid emoji code.
+    if emoji_type == "realm_emoji":
+        realm_emojis = realm.get_emoji()
+        if emoji_code in realm_emojis and not realm_emojis[emoji_code]["deactivated"]:
+            return True
+    elif emoji_type == "zulip_extra_emoji" and emoji_code == "zulip":
+        return True
+    elif emoji_type == "unicode_emoji" and emoji_code in codepoint_to_name:
+        return True
+    return False
+
+def emoji_name_is_valid(emoji_name, emoji_code, emoji_type):
+    # type: (Text, Text, Text) -> bool
```

Comment:
We no longer need to use `Text`; `str` is the same thing.  So for new code, I've been using `str` (and the new Python 3 syntax).  Since I was editing this (and the comments for it) anyway, I ended up just switching that.